src/shims/env.rs

```diff
@@ -370,15 +382,21 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             "`getcwd` is only available for the UNIX target family"
         );
 
-        this.check_no_isolation("`chdir`")?;
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("chdir", reject_with)?;
+            let err = Error::new(ErrorKind::NotFound, "rejected due to isolation");
+            this.set_last_error_from_io_error(err)?;
 
-        let path = this.read_path_from_c_str(this.read_scalar(path_op)?.check_init()?)?;
+            Ok(-1)
```

Comment:
Please return early in this branch, so that the code below does not have to be moved to the right.